platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,373 @@
+    - Does **NOT** run any content validation such as accessibility tests or
+      broken link checker
+2. [Full deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#full-deploy-of-vagov-client-app)
+    - For `vets-website` code changes
+    - Creates a new release and deploys it
+    - Fetches the latest Drupal content
+      - Falls back to the cached content in S3 from the last successful deploy
+        (full or partial)
+        <!-- TODO: Verify that either deploy will cache the pages.json and files -->
+    - Runs the accessibility tests and link checker
+
+See [here](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/)
+for more details.
+
+#### Pain points
+- Engineers encounter **confusing build errors** that are due to the mismatch in
+  outdated locally cached content and updated Liquid templates
+- Fetching content from the CMS and building HTML files when developing
+  applications **slows down the local builds** for engineers unnecessarily
+- When their applications have nothing to with the CMS
+- It **requires either SOCKS proxy access** or the `npm run fetch-drupal-cache`
+  command, which isn't obvious
+- **Application deployments and PR builds fail** when content in the CMS is
+  invalid
+  - This includes broken links and accessibility violations
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
+1. Content build
+    - Input: CMS content
+    - Output: HTML files, assets from the CMS (PDFs, images and the like)
+1. Application build
+    - Input: `vets-website` code
+    - Output: JavaScript and CSS bundles
+
+The output of both these builds will be deployed to separate buckets. The
+reverse proxies will route traffic intelligently to the appropriate bucket given
+the request.
+
+![Proposed traffic flow to S3 buckets](images/proposed-traffic-flow-to-s3-buckets.png)
+
+**Important note:** JavaScript application landing pages will live in the CMS.
+To get an application into production, the page in Drupal will need to be
+published.
+
+**Another important note:** There is no way to coordinate these two deploys to
+make an application live for the first time. The process will be to manually:
+1. Verify the application assets are live in production
+1. Publish the landing page
+1. Deploy a new content build
+
+## Specifics
+
+### Detailed Design
```

Comment:
I don't see anything about how these builds are triggered. when are these builds run and deployed? who can run/deploy them?

Reply:
Ah, good point. Short story:
- Application build is triggered on all commits pushed to GitHub (same as it is now)
- Application code is deployed on a weekdaily schedule
  - Probably in Jenkins at first until we can move it to Circle or something
- Content build and deploy (together) are triggered from Drupal
  - Same as it is now—I just don't know the specifics of how that all is wired up
  - This can be triggered by users with a certain permission in Drupal (again, not sure the specifics)

I'll sync with @ElijahLynn to get the specifics on the Drupal trigger and add those points in.

Reply:
Added a section on this below. It's pretty sparse, but it's what I know.